nsv13/code/modules/clothing/helmet_camera.dm

```diff
@@ -2,30 +2,25 @@
 	if(istype(I, /obj/item/wallframe/camera))
 		if(builtInCamera)
 			to_chat(user, "<span class='warning'>[src] already has a camera.</span>")
-			return
+			return FALSE
 		if(src == user.get_item_by_slot(ITEM_SLOT_HEAD)) //Make sure the player is not wearing the suit before applying the upgrade.
 			to_chat(user, "<span class='warning'>You cannot install the upgrade to [src] while wearing it.</span>")
-			return
-		if(user.transferItemToLoc(I, src))
-			builtInCamera = new /obj/machinery/camera(src)
-			builtInCamera.c_tag = "Helmet Cam #[rand(0,999)]"
-			builtInCamera.network = list("headcam", "ss13")
-			builtInCamera.internal_light = FALSE
-			QDEL_NULL(I)
+			return FALSE
+		if(do_after(install_camera(I, src), 5, target = src))
```

Comment:
The first argument of `do_after` is the user. The current call will immediately install the camera regardless of whether the do_after succeeds (and will probably runtime because the user is a `TRUE`).

Reply:
I forgot :p